src/EditorFeatures/Core/LanguageServer/AbstractInProcLanguageClient.cs

```diff
@@ -148,9 +148,7 @@ internal abstract partial class AbstractInProcLanguageClient : ILanguageClient,
 
             if (_languageServer is not null)
             {
-                Contract.ThrowIfFalse(_languageServer.HasShutdownStarted, "The language server has not yet been asked to shutdown.");
-
-                await _languageServer.DisposeAsync().ConfigureAwait(false);
+                await _languageServer.WaitForExitAsync().ConfigureAwait(false);
```

Comment:
This may hang if no exit notifications are sent/the exit notification fails. Should we supply a timeout?

Reply:
the caller passes a cancellation token which I should be respecting here.  good catch.